evaluator/evaluator/routers/predictor.py

```diff
@@ -0,0 +1,110 @@
+import datetime
+import json
+import logging
+from typing import Dict, List
+
+import requests
+from bson import ObjectId
+from fastapi import APIRouter, Depends, Request
+from fastapi.exceptions import HTTPException
+from square_auth.auth import Auth
+
+from evaluator import mongo_client
+from evaluator.core import DatasetHandler
+from evaluator.core.dataset_handler import DatasetDoesNotExistError
+from evaluator.models import (
+    DatasetResult,
+    Prediction,
+    PredictionAnswer,
+    ReferenceAnswer,
+)
+from evaluator.routers import client_credentials
+
+logger = logging.getLogger(__name__)
+router = APIRouter(prefix="/predictor")
+auth = Auth()
+
+
+@router.post(
+    "/{skill_id}/{dataset_name}",
+    status_code=201,
+)
+async def predict(
+    request: Request,
+    skill_id: str,
+    dataset_name: str,
+    dataset_handler: DatasetHandler = Depends(DatasetHandler),
+    token: str = Depends(client_credentials),
+    _token_payload: Dict = Depends(auth),
+):
+    logger.debug(
+        f"start predictions with parameters: skill_id={skill_id}; dataset_name={dataset_name}"
+    )
+    try:
+        dataset = dataset_handler.get_dataset(dataset_name)
+    except DatasetDoesNotExistError:
+        logger.debug("Dataset does not exist!")
+        raise HTTPException(400, "Dataset does not exist!")
+    logger.debug(f"Dataset loaded: {dataset}")
+
+    headers = {"Authorization": f"Bearer {token}"}
+    if request.headers.get("Cache-Control"):
+        headers["Cache-Control"] = request.headers.get("Cache-Control")
+
+    predictions: List[Prediction] = []
+    start_time = datetime.datetime.now()
+
+    for i in range(8):
+        reference_data = dataset[i]
+        query_request = {
+            "query": reference_data["question"],
+            "skill_args": {"context": reference_data["context"]},
```

Comment:
are this fields valid for all datasets in the dataset list? Ex: https://huggingface.co/datasets/commonsense_qa doest have a context field, https://huggingface.co/datasets/hotpot_qa has

Reply:
How to handle DatasetDict in future? Datasethandler returns either Dataset or DatasetDict